core/src/test/java/org/infinispan/test/fwk/TestCacheManagerFactory.java

```diff
@@ -72,7 +72,6 @@
    public static EmbeddedCacheManager newDefaultCacheManager(boolean start, GlobalConfigurationBuilder gcb,
                                                              ConfigurationBuilder c) {
       setNodeName(gcb);
-      MetricsCollectorTestHelper.replace(gcb);
```

Comment:
I am not familiar with the difference of micrometer and microprofile metrics. But are we isolating the former's registry or something now? And are we controlling its lifecycle with the cache? I didn't see code for that.

Reply:
`MetricsCollector` contains the registry implementation and avoids exposing it outside.
Some metrics (`BaseAdditionalMetrics` and `VendorAdditionalMetrics`) are added when the instance of `MetricsCollector` is created by its factory.
The others metrics are added by `AbstractMetricsRegistration#processComponents`.

Reply:
The first process is triggered (I think) by the `@Start` annotation of `MetricsCollector`,
The second process is triggered (I always just think) by the `@Start` of `AbstractMetricsRegistration`.

Reply:
I think we need to invoke https://github.com/micrometer-metrics/micrometer/blob/bdac8287daf1ff8c297648056a06fc1f8613f04f/micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java#L1028 in a @Stop method as well to clean up those registries.